packages/xdl/src/Project.js

```diff
@@ -1138,9 +1143,19 @@ export async function buildAsync(
     throw new XDLError(ErrorCode.INVALID_OPTIONS, e.toString());
   }
 
-  let { exp, pkg } = await ProjectUtils.readConfigJsonAsync(projectRoot);
-  const configName = await ProjectUtils.configFilenameAsync(projectRoot);
-  const configPrefix = configName === 'app.json' ? 'expo.' : '';
+  let exp, pkg, configName, configPrefix;
+  if (!options.publicUrl) {
+    // get the manifest from the project directory
+    ({ exp, pkg } = await ProjectUtils.readConfigJsonAsync(projectRoot));
+    const configName = await ProjectUtils.configFilenameAsync(projectRoot);
```

Comment:
i guess it's redundant `const`

Reply:
good catch. I decided to move all this logic to a different function so `exp`, `pkg`, `configName` and `configPrefix` can all be `const`